rule/rule.go

```diff
@@ -369,9 +369,18 @@ func (f *File) Format() []byte {
 	return bzl.Format(f.File)
 }
 
+// Sort rules in this File
+func (f *File) Sort() {
+	//TODO: sort load statements too
+	if f.function != nil {
+		sort.Stable(byName{f.Rules, f.function.stmt.Body})
+	}
+}
+
 // Save writes the build file to disk. This method calls Sync internally.
 func (f *File) Save(path string) error {
 	f.Sync()
+	f.Sort()
```

Comment:
If `f.Sort()` is called unconditionally here, won't that sort `WORKSPACE` and `BUILD.bazel` files? I think only dependency macros should be sorted.

Reply:
`f.Sort()` is a no-op if `f.function == nil`

Reply:
Ah, sorry, I missed that.

Based on the method signatures and documentation, this behavior seems surprising to me. I think that:

* `Sort` should work unconditionally.
* The caller in `update-repos.go` should be responsible for calling `Sort`. This package doesn't know whether it's appropriate to call `Sort`; it may be controlled by directives or something else later on.
* `Sort` should internally call `Sync` (and the doc should mention that). It's okay if `Sync` is called again later in `Save`.